File: solutions/java/rotational-cipher/1/src/main/java/RotationalCipher.java

class RotationalCipher {

    int shiftKey;

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {

        StringBuilder rotated = new StringBuilder();

        data.chars().forEach(c -> {
            rotated.append(
                    rotateCharacter(c));
        });

        return rotated.toString();
    }

    char rotateCharacter(int c) {

        if (Character.isAlphabetic(c)) {
            char base = Character.isUpperCase(c) ? 'A' : 'a';

            return (char) ((((c - base) + shiftKey) % 26) + base);
        }

        return (char) c;

    }

}

File: solutions/csharp/rotational-cipher/1/RotationalCipher.cs
public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey) => string
        .Join("", text.Select(c => Rotate(c, shiftKey)));

    private static char Rotate(char c, int shiftKey)
    {
        if (!IsLetter(c)) return c;

        var offset = c is >= 'a' and <= 'z' ? 'a' : 'A';

        var value = c - offset;
        
        var nextValue = (value + shiftKey) % 26;
        
        return (char)(nextValue + offset);

    }
    
    private static bool IsLetter(char c) => c is >= 'A' and <= 'Z' or >= 'a' and <= 'z';
}